src/client/Microsoft.Identity.Client.Brokers/RuntimeBroker.cs

```diff
@@ -0,0 +1,392 @@
+                        _logger.Verbose("[WamBroker] Successfully retrieved token.");
+
+                    }
+                    else
+                    {
+                        _logger.Error($"[WamBroker] Could not login interactively. {result.Error}");
+                        WamAdapters.CreateWamErrorResponse(result, authenticationRequestParameters, _logger);
+                    }
+                }
+            }
+            
+            return msalTokenResponse;
+        }
+
+        /// <summary>
+        /// AcquireToken Interactively for the default user using WAM
+        /// </summary>
+        /// <param name="authenticationRequestParameters"></param>
+        /// <param name="acquireTokenInteractiveParameters"></param>
+        /// <returns></returns>
+        /// <exception cref="MsalServiceException"></exception>
+        public async Task<MsalTokenResponse> AcquireTokenInteractiveDefaultUserAsync(
+            AuthenticationRequestParameters authenticationRequestParameters,
+            AcquireTokenInteractiveParameters acquireTokenInteractiveParameters)
+        {
+            MsalTokenResponse msalTokenResponse = null;
+
+
+            var cancellationToken = authenticationRequestParameters.RequestContext.UserCancellationToken;
+
+            _logger.Verbose("[WamBroker] Signing in with the default user account.");
+
+            using (var core = new NativeInterop.Core())
+            using (var authParams = WamAdapters.GetCommonAuthParameters(authenticationRequestParameters, _wamOptions.MsaPassthrough))
+            {
+                using (NativeInterop.AuthResult result = await core.SignInAsync(
+                        _parentHandle,
+                        authParams,
+                        authenticationRequestParameters.CorrelationId.ToString("D"),
+                        cancellationToken).ConfigureAwait(false))
+                {
+                    if (result.IsSuccess)
+                    {
+                        msalTokenResponse = WamAdapters.ParseRuntimeResponse(result, authenticationRequestParameters, _logger);
+                    }
+                    else
+                    {
+                        _logger.Error($"[WamBroker] Could not login interactively with the Default OS Account. {result.Error}");
+                        WamAdapters.CreateWamErrorResponse(result, authenticationRequestParameters, _logger);
+                    }
+                }
+            }
+
+            return msalTokenResponse;
+        }
+
+        /// <summary>
+        /// Gets the window handle
+        /// </summary>
+        /// <param name="uiParent"></param>
+        /// <returns></returns>
+        private IntPtr GetParentWindow(CoreUIParent uiParent)
+        {
+            if (uiParent?.OwnerWindow is IntPtr ptr)
+            {
+                return ptr;
+            }
+
+            return IntPtr.Zero;
+        }
+
+        /// <summary>
+        /// AcquireTokenSilentAsync
```

Comment:
Nit: maybe this for consistency
```suggestion
        /// Acquire Token Silently
```